### ISSUE TITLE
Unreadable text in Firefox using hc-black theme with Windows HC White

### ISSUE THREAD

**Issue by jayspadie** [Microsoft/monaco-editor#407]
monaco-editor version: 0.8.3
Browser:  Firefox
OS: Windows 10

During accessibility testing, it was found that if the user sets their **Windows theme to High Contrast White**, and **hc-black** is used as the theme in Monaco, then the result in Firefox is white text on a white background, making it unreadable.  In other browsers (Edge, IE, Chrome) Monaco uses the hc-black theme provided.  Examples from the [Monaco playground](https://microsoft.github.io/monaco-editor/playground.html#creating-the-editor-editor-basic-options) below.

This is an issue specific to Firefox where it seems to more aggressively render black text on a white background with blue hyperlinks, and it doesn't play well with hc-black.  However, if the default theme is used in Monaco, then the text is rendered correctly as black on white.  So, if it's possible in JavaScript to detect that the user is using the Windows High Contrast White theme and Firefox, then one fix would be to switch to use the default theme in Monaco.

**Chrome**:  (looks fine)
![image](https://cloud.githubusercontent.com/assets/5246869/24626231/94ac7d6c-187e-11e7-854e-95d35ae04cbf.png)

**Firefox**: (unreadable white on white)
![image](https://cloud.githubusercontent.com/assets/5246869/24626267/be12f726-187e-11e7-9a0b-b99b7d3c2c05.png)

**Edge**: (looks fine)
![image](https://cloud.githubusercontent.com/assets/5246869/24626309/d8cc15c0-187e-11e7-919c-c4f0e8d06805.png)

**Comment by alexandrudima** [comment]
The root problem here is that we don't ship a white HC theme in VS Code and all changes disregard this theme. @aeschli do we still have this theme with our theme rework or was it completely removed?

**Comment by aeschli** [comment]
@alexandrudima The hc_white is gone since a long time. I never saw it in action.

**Comment by alexandrudima** [comment]
oh disregard that, this is the hc-black theme

**Comment by alexandrudima** [comment]
This might be a windows only issue, I cannot reproduce on FF on OSX

**Comment by alexandrudima** [comment]
Ok, after reading in more detail, I can reproduce. The minimal steps include:
* set the Windows Theme to High Contrast White
* load up an editor with `hc-black`
* Firefox refuses to paint background colors, tokens colors, which somehow culminates with white tokens being painted on top of white background. I'm not sure why they end up doing this at this point.


Under Chrome, we function correctly because, most likely, Chrome ignores the Windows Theme.

Under IE/Edge, we function correctly because we use `-ms-high-contrast-adjust: none;`, to indicate to Trident that we have picked our own HC colors responsibly.

Under Firefox, which respects the Windows Theme, we don't function correctly because there is no equivalent of `-ms-high-contrast-adjust`. 

Here is the upstream Firefox issue that tracks this - https://bugzilla.mozilla.org/show_bug.cgi?id=957988 -- I have searched their issue tracker for some time and could not find anything else that would allow us to begin working around. i.e. there is no way to know in Firefox when the Windows Theme is a high contrast theme.

**Comment by alexandrudima** [comment]
With the removal of  `mix-blend-mode: difference` for `hc-black` (https://github.com/Microsoft/vscode/commit/ab5846a2ab4a25df02997de7e894582521d5a532), the high contrast theme renders black text on Firefox, when Windows is set to High Contrast White. Various things are not great, such as the editor selection is effectively invisble (which uses a background color that Firefox refuses to paint).

![image](https://user-images.githubusercontent.com/5047891/27473555-df0c58da-57ff-11e7-89dd-6db921f66445.png)
